### DISCUSSION TITLE
Redirect to Privacy respecting alternatives

### DISCUSSION THREAD

**Discussion by GreenLunar** [searxng/searxng#971] [Q&A]
In the spirit of ticket #821 and UntrackMe

Realization
https://f-droid.org/en/packages/app.fedilab.nitterizeme/
https://fedilab.app/wiki/nitterizeme/

>  UntrackMe once set by default to open Twitter, YouTube and Instagram links will transform them into Nitter, Invidious and Bibliogram links and delegate the action to another app.
> It also supports Google Maps links to transform them into OpenStreetMap links or Geo URI.
> The app will detect shortened URLs and will display the full URL before visiting the site.

> **Lite version:
> - When sharing a Twitter, YouTube or Google Maps link just pick up UntrackMe to transform it before sharing it with another app.

> **Full version:
> - Contains all features of the lite version and it also handle all links to remove tracking parameters. It uses its own app picker that display URLs before visiting them.

> What is Nitter:
> A free and open source alternative Twitter front-end focused on privacy.
Github project: https://github.com/zedeus/nitter

> What is Invidious:
> A free and open source alternative front-end to YouTube
Github project: https://github.com/omarroth/invidious

> What is Bibliogram:
> A free and open source alternative front-end to Instagram
Github project: https://github.com/cloudrac3r/bibliogram

> What is Teddit:
> A free and open source alternative Reddit front-end focused on privacy.
Github project: https://github.com/teddit-net/teddit

> What is Scribe:
> An alternative Medium front-end
Sources: https://git.sr.ht/~edwardloveall/scribe

> What is Wikiless:
> A free open source alternative Wikipedia front-end focused on privacy.
Project: https://codeberg.org/orenom/Wikiless

**Comment by return42** [comment]
Just wanted to add a link to https://github.com/searxng/searxng/issues/725

**Comment by dalf** [comment]
Side note:
* A link to another project: https://github.com/benbusby/farside
* The list is auto updated: https://github.com/benbusby/farside/blob/main/.github/workflows/update-instances.yml

**Comment by emanruse** [comment]
It would be good onion instances of Searx to use onion instances of the alternatives.

Please consider also:

https://github.com/mendel5/alternative-front-ends


Facebook doesn't seem to have an alternative front-end. However it has a version which doesn't need JavaScript:

https://mbasic.facebook.com/

And an official onion instance too:

https://mbasic.facebookwkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd.onion/

**Comment by peepo5** [comment]
farside seems like a great gateway but then how do you know how much strain it would put on the instance?

**Comment by GreenLunar** [comment]
Some instances already got links to "other instances", so it's feasible
to make an automatic redirect mechanism that would be activiated when
needed.

Such mechanism needs to be added to each instance seperately.

**Comment by return42** [comment]
> Such mechanism needs to be added to each instance separately.

That's right, there isn't always a generic solution that fits to the needs. We can discuss the various aspects in our Q&A section:

- https://github.com/searxng/searxng/discussions/911
- https://github.com/searxng/searxng/discussions/970

I will also move this Issue in the Q&A section.

**Comment by ghost** [comment]
[Libredirect](https://github.com/libredirect/libredirect) is a great browser plugin that redirects users to privacy respecting alternatives, and I've been thinking about writing a searxng plugin similar to hostname_replace that does all of the replacements that libredirect does. But I don't really want to maintain it.

My current hostname_replace configuration redirects:

* youtube to the invidious instance that I use
* reddit to the libreddit instance that I use
* twitter to a nitter instance
* wikipedia to wikiless.org
* imgur to a rimgo instance
* instagram to my own bibliogram instance

I was hoping that someone would have shared their hostname_replace settings in this thread already, but I guess I'll make a start and hope that others will add to it:

```yaml
hostname_replace:
  '(.*\.)?youtube\.com$':           'invidious.kavin.rocks'
  '(.*\.)?youtu\.be$':              'invidious.kavin.rocks'
  '(.*\.)?youtube-noocookie\.com$': 'invidious.kavin.rocks'
  '(.*\.)?reddit\.com$':            'libreddit.kavin.rocks'
  '(.*\.)?redd\.it$':               'libreddit.kavin.rocks'
  '(www\.)?twitter\.com$':          'nitter.snopyta.org'
  '(.*\.)?wikipedia.org$':          'wikiless.org'
  '(i\.)?imgur.com$':               'i.bcow.xyz'
  '(www\.)?instagram.com$':          'bibliogram.pussthecat.org' # I actually use a private instance here
```

I'd like to include scribe for medium.com but it only works with certain pages. I'd also like to be able to use an array of hostnames in hostname_replace so that I'm not always using the same instances for everything.

**Reply by sjehuda** [reply]
It's interetsing that your comment comes a few days after I've published a Userscript, with the generous help from the OP, that does just that. https://openuserjs.org/scripts/sjehuda/Farside

A redirector that behaves like Farside would be great
> [Farside](https://farside.link/) provides links that automatically redirect to working instances of privacy-oriented alternative frontends, such as Nitter, Libreddit, etc. This allows for users to have more reliable access to the available public instances for a particular service, while also helping to distribute traffic more evenly across all instances and avoid performance bottlenecks and rate-limiting.

**Reply by sjehuda** [reply]
Link update to the userscript https://openuserjs.org/scripts/sjehuda/FREEdirector

**Comment by return42** [comment]
> I was hoping that someone would have shared their hostname_replace settings in this thread already, but

Above I linked the discussion : https://github.com/searxng/searxng/discussions/970 .. I updated the list two weeks ago .. feel welcome to add your settings to the discussion / thanks!

**Comment by emanruse** [comment]
> A redirector that behaves like Farside would be great

Exposing any form of personal data to additional mid-man reduces privacy. The links should lead directly to the privacy respecting alternatives.
